features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/net/BatchesToSegmentsMapperTest.kt

```diff
@@ -48,158 +48,150 @@ internal class BatchesToSegmentsMapperTest {
-        val rumContext: SessionReplayRumContext = forge.getForgery()
         val fakeRecordsSize = forge.anInt(min = 10, max = 20)
         val fakeTimestamp = forge.aTimestamp()
         val fakeRecords: List<MobileSegment.MobileRecord> = forge.aList(fakeRecordsSize) {
             forge.getForgery<MobileSegment.MobileRecord>().copy(fakeTimestamp)
         }
         val fakeEnrichedRecords = fakeRecords
-            .chunked(forge.anInt(min = 1, max = fakeRecordsSize))
-            .map {
-                EnrichedRecord(
-                    rumContext.applicationId,
-                    rumContext.sessionId,
-                    rumContext.sessionId,
-                    it
-                )
-            }
-        val fakeExpectedEnrichedRecord = EnrichedRecord(
-            rumContext.applicationId,
-            rumContext.sessionId,
-            rumContext.sessionId,
-            fakeRecords
-        )
-
-        val expectedEmptySegment =
-            fakeExpectedEnrichedRecord.toSegment().copy(records = emptyList())
-        val expectedSerializedSegment = fakeExpectedEnrichedRecord.toSegment()
-            .toJson().asJsonObject.toString()
+                .chunked(forge.anInt(min = 1, max = fakeRecordsSize))
+                .map {
+                    val rumContext: SessionReplayRumContext = forge.getForgery()
+                    EnrichedRecord(
+                            rumContext.applicationId,
+                            rumContext.sessionId,
+                            rumContext.sessionId,
+                            it
+                    )
+                }
+        val fakeExpectedPairs = fakeEnrichedRecords.map {
+            Pair(it.toSegment().copy(records = emptyList()), it.toSegment().toJson())
+        }
         val fakeBatchData = fakeEnrichedRecords.map { it.toJson().toByteArray() }
 
         // When
-        val mappedSegment = testedMapper.map(fakeBatchData)
+        val mappedSegments = testedMapper.map(fakeBatchData)
 
         // Then
-        val segment = mappedSegment?.first
-        val serializedSegment = mappedSegment?.second.toString()
-        assertThat(segment).isEqualTo(expectedEmptySegment)
-        assertThat(serializedSegment).isEqualTo(expectedSerializedSegment)
+        mappedSegments.forEachIndexed { index, pair ->
+            assertThat(pair.first).isEqualTo(fakeExpectedPairs[index].first)
+            assertThat(pair.second.toString()).isEqualTo(fakeExpectedPairs[index].second.toString())
+        }
     }
 
     @Test
     fun `M use the first record in the list as start timestamp W map`(
         forge: Forge
     ) {
-        // Given
-        val rumContext: SessionReplayRumContext = forge.getForgery()
         val fakeRecordsSize = forge.anInt(min = 10, max = 20)
-        val fakeRecords: List<MobileSegment.MobileRecord> = forge.aList(fakeRecordsSize) {
+        val fakeRecords: List<MobileSegment.MobileRecord> = forge.aList<MobileSegment.MobileRecord>(fakeRecordsSize) {
```

Comment:
minor: the same on the line L54 doesn't require type to be present twice. Same for L161.